### PR TITLE
Adding some rounding and error messaging for subplots

### PR DESCRIPTION
**Pull request by stephenpardy**
## Code PR

- [x] I have read through the [contributing notes](https://github.com/plotly/plotly.py/blob/master/contributing.md) and understand the structure of the package. In particular, if my PR modifies code of `plotly.graph_objects`, my modifications concern the `codegen` files and not generated files.
- [x] I have added tests (if submitting a new feature or correcting a bug) or
  modified existing tests.
- [ ] For a new feature, I have added documentation examples in an existing or
  new tutorial notebook (please see the doc checklist as well).
- [x] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.
- [ ] For a new feature or a change in behaviour, I have updated the relevant docstrings in the code to describe the feature or behaviour (please see the doc checklist as well).

-->

### CODE REVIEW

**Comment by stephenpardy** [comment]
@AaronStiff - this is the PR to close #4124 - I mostly moved over your code here. I am seeing an NPM error in the CI/CD process though. Would you be available to review?

**Comment by AaronStiff** [comment]
@stephenpardy Great! I'm not really familiar with/involved in the build process though, but maybe @nicolaskruchten you could take a look?

**Comment by nicolaskruchten** [comment]
@alexcjohnson can you review this one plz? I'm not really available for the next few days

**Comment by stephenpardy** [comment]
@AaronStiff, @nicolaskruchten, or @alexcjohnson - just want to give this a gentle nudge to see if anyone is able to review and/or discuss the NPM build failure - I see that it is still around in some other PRs as well. Thank you.

File: packages/python/plotly/plotly/_subplots.py [hunk 1]
```diff
@@ -702,6 +702,26 @@ def _check_hv_spacing(dimsize, spacing, name, dimvarname, dimname):
...
+                        "A combination of the 'b' values, heights, and "
+                        "number of subplots too large for this subplot gird."
+                    )
+            if y_e > 1.0:
+                if y_e < 1.01:
+                    y_e = 1.0
+                else:
+                    raise Exception(
+                        "A combination of the 't' values, heights, and "
+                        "number of subplots too large for this subplot gird."
```
Comment (alexcjohnson) [inline packages/python/plotly/plotly/_subplots.py:722]:
```suggestion
                        "A combination of the 'b' values, heights, and "
                        "number of subplots too large for this subplot grid."
                    )
            if y_e > 1.0:
                if y_e < 1.01:
                    y_e = 1.0
                else:
                    raise Exception(
                        "A combination of the 't' values, heights, and "
                        "number of subplots too large for this subplot grid."
```

**Comment by alexcjohnson** [comment]
Thanks @stephenpardy - we did some other work in the last few days to fix build/CI issues so with any luck the tests will pass now that I updated the branch. Would you be able to add a test of the problem you're fixing, as well as an entry in the changelog?

**Review by alexcjohnson — APPROVED** [review]
💃 Beautiful, thank you!

**Pull request merged**